### PR TITLE
Create navigation header

### PR DESCRIPTION
**Pull request by AllStackDev1**
Closes: #1362
# Create navigation header

## Background

As a user, I should see a header with the GitHub logo as a home button on the left, and my user profile picture on the right attached to a dropdown menu. The menu should show links for navigating to my profile and a sign out link.

## Acceptance

- [x] Top nav
  - [x] GitHub logo on the left
  - [x] User avatar with dropdown on the right
  - [x] Navigation to profile in the dropdown (can point to not yet existing page)
  - [x] Logout link in the dropdown (can point to not yet existing functionality yet)
- [x] Placeholder for main content

## Notes

### CODE REVIEW

File: solidstart-tanstackquery-tailwind-modules/src/components/UserDropdown/UserDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  image: string;
+  username: string;
+}
+
+const UserDropdown = (props: IProps) => {
+  const [expanded, setExpanded] = createSignal(false);
+
+  const _clickOutside = clickOutside;
+
+  const signOut = () => {
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/UserDropdown/UserDropdown.tsx:19]: I believe there is a ticket for sign-out functionality. This shouldn't have been implemented yet

File: solidstart-tanstackquery-tailwind-modules/src/components/UserDropdown/UserDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          </div>
+          <ChevronDownIcon class="-mr-1 ml-2 h-3 w-3" aria-hidden="true" />
+        </button>
+        <nav
+          class={
+            expanded() ? styles.dropdownMenuBase : styles.dropdownMenuHidden
+          }
+        >
+          <ul class="py-1">
+            {props.username && (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/UserDropdown/UserDropdown.tsx:60]: make use of `switch and match` or `show` for solidjs here. We need to be more solidjs than react

**Review by hdJerry — CHANGES_REQUESTED** [review]

File: solidstart-tanstackquery-tailwind-modules/src/utils/onclickOutside.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { onCleanup } from 'solid-js';
+
+export const clickOutside = (el: any, accessor: any) => {
+  const onClick = (e: any) => {
+    !el.contains(e.target) && accessor()?.();
+  };
+  document.body.addEventListener('click', onClick);
+
+  onCleanup(() => document.body.removeEventListener('click', onClick));
+};
```
Comment (Megio) [inline solidstart-tanstackquery-tailwind-modules/src/utils/onclickOutside.ts:10]: At least we can typed the el and e, never see accessor, don't know what type it is :-D

```suggestion
export const clickOutside = (el: HTMLDivElement, accessor: any) => {
  const onClick = (e: MouseEvent) => {
    !el.contains(e.target as unknown as HTMLDivElement) && accessor()?.();
  };
  document.body.addEventListener('click', onClick);

  onCleanup(() => document.body.removeEventListener('click', onClick));
};
```

File: solidstart-tanstackquery-tailwind-modules/src/components/UserDropdown/UserDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    // @ts-ignore
+    <div use:_clickOutside={() => setExpanded(false)}>
+      <div class={styles.dropdown}>
+        <button
+          role="button"
+          class={styles.dropdownBtn}
+          onClick={() => setExpanded(!expanded())}
+        >
+          <div class={styles.avatarContainer}>
+            {props.image && (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/UserDropdown/UserDropdown.tsx:27]: you forgot Show here 😀

**Review by hdJerry — CHANGES_REQUESTED** [review]

**Review by hdJerry — APPROVED** [review]
Thanks sir

**Pull request merged**